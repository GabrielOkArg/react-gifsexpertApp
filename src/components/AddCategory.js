import React, { useState } from 'react';
import PropTypes from 'prop-types';



export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState('');

    const handleInputValue =(e)=>{
       setinputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        setCategories(cate => [inputValue,...cate]);
        setinputValue('');
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                placeholder='Que gif estas buscando?'
                value={inputValue}
                onChange={handleInputValue}
                required
            />
        </form>
    );
   
}

AddCategory.protoTypes={
    setCategories: PropTypes.func.isRequired
}
