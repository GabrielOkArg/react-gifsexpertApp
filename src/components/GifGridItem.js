import React from 'react';
import PropTypes from 'prop-types'; //impt

//import '../index.css';

export const GifGridItem = ({title,url}) => {

    return (
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
        );
};

GifGridItem.protoType={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}
