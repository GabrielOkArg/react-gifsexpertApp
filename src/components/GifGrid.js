import React, { useState, useEffect } from 'react';
import { useFecthGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({ category }) => {

    const {data:images,lodading} = useFecthGifs(category);

    return (
        <>
             <h3>{category}</h3>
           {/* el operador && sirve para no hacer un ternario no evalua el false */}
           {lodading && <p>Loading</p>} 
            <div className='card-grid'>
                {
                    images.map((item) => {
                        return <GifGridItem key={item.id} {...item} />
                    })
                }
            </div>
        </>
    );
};
