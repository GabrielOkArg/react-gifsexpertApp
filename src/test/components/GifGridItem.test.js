import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';




describe('Pruebe de GitGifItem', () => {

    const url = 'https://media2.giphy.com/media/FRkBbAdRf2jTfahZaA/giphy.gif?cid=22b511629gb4qvrh6sj1zig5w834he0677hd19mn0pmmyg8s&rid=giphy.gif&ct=g';
    const title = 'Dragon Ball GIF by TOEI Animation UK';

    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test('debe mostrar un item correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un <p> </p> con el title', () => {
      const p = wrapper.find('p');
      expect(p.text().trim()).toBe(title);
    });

    test('Debe tener la imagen igual al rul y alt de los props', () => {
      const img = wrapper.find('img');
      expect(img.prop('src')).toBe(url);
      expect(img.prop('alt')).toBe(title);
      
    });
    
    test('Debe de tener animate__fadein', () => {
      const div = wrapper.find('div');
      //console.log(div.props().className);
      expect(div.prop('className')).toContain('animate__fadeInDown')
    });
    

});
