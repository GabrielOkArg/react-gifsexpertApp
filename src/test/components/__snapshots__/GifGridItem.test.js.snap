// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebe de GitGifItem debe mostrar un item correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Dragon Ball GIF by TOEI Animation UK"
    src="https://media2.giphy.com/media/FRkBbAdRf2jTfahZaA/giphy.gif?cid=22b511629gb4qvrh6sj1zig5w834he0677hd19mn0pmmyg8s&rid=giphy.gif&ct=g"
  />
  <p>
    Dragon Ball GIF by TOEI Animation UK
  </p>
</div>
`;
